### PR TITLE
Tmpdir fix for cleanup testfiles

### PR DESCRIPTION
**Pull request by desikan95**
Preventing temporary files from being created in the working directory. Uses Pytest's tmpdir fixture to handle this.

### CODE REVIEW

File: tests/integration_tests/test_experiment.py [hunk 1]
```diff
@@ -327,12 +329,13 @@ def test_experiment_image_inputs(image_parms: ImageParms, csv_filename: str):
     input_features[0]['encoder'] = image_parms.image_encoder
     input_features[0]['preprocessing'][
         'in_memory'] = image_parms.in_memory_flag
-    rel_path = generate_data(input_features, output_features, csv_filename)
+    rel_path = generate_data(input_features, output_features, os.path.join(tmpdir,csv_filename))
```
Comment (jimthompson5802) [inline tests/integration_tests/test_experiment.py:332]: The project is following PEP8 style guidelines.  In particular there should be a space following a comma, e.g., `os.path.join( tmpdir, csv_filename)`

File: tests/integration_tests/test_experiment.py [hunk 2]
```diff
@@ -348,12 +351,13 @@ def test_experiment_image_inputs(image_parms: ImageParms, csv_filename: str):
...
         train_format, train_in_memory,
-        test_format, test_in_memory
+        test_format, test_in_memory,
+        tmpdir
 ):
     # primary focus of this test is to determine if exceptions are
     # raised for different data set formats and in_memory setting
     # Image Inputs
-    image_dest_folder = os.path.join(os.getcwd(), 'generated_images')
+    image_dest_folder = os.path.join(os.getcwd(),tmpdir+'generated_images')
```
Comment (jimthompson5802) [inline tests/integration_tests/test_experiment.py:360]: combining `os.getwd()` and `tmpdir` to form may not be the correct approach.

File: tests/integration_tests/test_experiment.py [hunk 3]
```diff
@@ -386,8 +390,8 @@ def test_experiment_image_dataset(
     }
 
     # create temporary name for train and test data sets
-    train_csv_filename = 'train_' + uuid.uuid4().hex[:10].upper() + '.csv'
-    test_csv_filename = 'test_' + uuid.uuid4().hex[:10].upper() + '.csv'
+    train_csv_filename = tmpdir+'train_' + uuid.uuid4().hex[:10].upper() + '.csv'
+    test_csv_filename = tmpdir+'test_' + uuid.uuid4().hex[:10].upper() + '.csv'
```
Comment (jimthompson5802) [inline tests/integration_tests/test_experiment.py:394]: In theory `tmpdir + 'test_' + ....' may or may not work depending if `tmpdir` has a trailing slash.  A better approach is to use the `os.path.join()` function to create the fully qualified path name.  This makes it platform independent.

File: tests/integration_tests/test_experiment.py [hunk 4]
```diff
@@ -142,35 +142,36 @@ def test_experiment_seq_seq_model_def_file(csv_filename, yaml_filename):
...
-        input_features, output_features, 'tr_' + csv_filename
+        input_features, output_features, os.path.join(tmpdir,'tr_'+csv_filename)
     )
     test_csv = generate_data(
-        input_features, output_features, 'test_' + csv_filename, 20
+        input_features, output_features, os.path.join(tmpdir,'test_'+csv_filename), 20
     )
     valdation_csv = generate_data(
-        input_features, output_features, 'val_' + csv_filename, 20
+        input_features, output_features,os.path.join(tmpdir,'val_'+csv_filename), 20
```
Comment (jimthompson5802) [inline tests/integration_tests/test_experiment.py:165]: Note the project's use of whitespace around binary operators.  PEP8 guidelines.

File: tests/integration_tests/test_experiment.py (hunk 2 above)
Comment (tgaddair) [reply tests/integration_tests/test_experiment.py:360]: Yeah, I would ditch the cwd stuff and consider creating a new tempdir to store the images:

```
with tempfile.TemporaryDirectory() as image_dest_folder:
    ...
```

**Pull request closed without merging**